src/app/task.service.ts

```diff
@@ -1,17 +1,23 @@
 import { Injectable } from '@angular/core';
-import * as TaskJson from './task_sample_data.json';
+//import * as TaskJson from './task_sample_data.json';
 import { TaskData, TaskList } from './taskdata';
 import { BehaviorSubject, Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { catchError } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TaskService {
 
-  taskListSubject: BehaviorSubject<TaskData[]> = new BehaviorSubject<TaskData[]>(TaskJson.tasks);
+  private taskListSubject: BehaviorSubject<TaskData[]> = new BehaviorSubject<TaskData[]>([]);
 
   taskList$: Observable<TaskData[]> = this.taskListSubject.asObservable();
 
+  constructor(private http: HttpClient) {
+    this.loadTasksFromAPI();
```

Comment:
try to avoid doing any kind of subscriptions in the service as it is hard to manage the subscription from here, instead you should call this method when you open the task list page or something